lib/OpenQA/Worker/Settings.pm

```diff
@@ -45,6 +47,15 @@ sub new ($class, $instance_number = undef, $cli_options = {}) {
     my @hosts = split(' ', $cli_options->{host} || $global_settings{HOST} || 'localhost');
     delete $global_settings{HOST};
     _read_section($cfg, $_, $webui_host_specific_settings{$_} = {}) for @hosts;
+    my %tokens = map {
+        my ($host, $key, $secret) = decode_token($_);
+        $host => {key => $key, secret => $secret}
+    } split /,/, $ENV{OPENQA_WORKER_TOKEN} // $cli_options->{token} // '';
+    for (keys %tokens) {
+        $webui_host_specific_settings{$_} ||= {};
```

Comment:
```suggestion
```

Reply:
so you think this line isn't necessary? Isn't that needed for auto-vivification so that we can write values into the inner hash?